src/common/Result.h

```diff
@@ -79,9 +79,6 @@ class NXT_NO_DISCARD Result<void, E*> {
 template <typename T, typename E>
 class NXT_NO_DISCARD Result<T*, E*> {
   public:
-    static_assert(alignof(T) >= 4, "Result<T*, E*> reserves two bits for tagging pointers");
-    static_assert(alignof(E) >= 4, "Result<T*, E*> reserves two bits for tagging pointers");
-
```

Comment:
The problem is that alignof needs to "see" the definition of the type. This is a problem because ideally the definition of ErrorData is only known by helper functions and Device's error handling code (to avoid having `<string>` and `<vector>` included everywhere). WDYT?

Reply:
Aha. Is there any way to assert `!is_concrete(T) || alignof(T) >= 4`?